### PR TITLE
chore: matic -> pol updates

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Updates hardcoded instances of MATIC to POL.

Note, TODOs left on the Ramps because we need to confirm if/when/how they update these values upstream so we can map to them.

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/7620

## Risk

> High Risk PRs Require 2 approvals

Medium

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

All interactions with Polygon chain.

## Testing

Ensure:

- On-ramping to Polygon works across all chains (pending confirmation with each ramp closer to the time)
- Trading to and from assets on the Polygon chain works, with gas taken from the POL token on Polygon

### Engineering

☝️

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

☝️

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xean** [comment]
@shapeshift/operations - we expect that at some point in the future the on ramps will break as they transition to the correct name. Can you help us to check occasionally on polygon specifically with these providers so we can fix it once they migrate? 

cc @purelycrickets

**Comment by 0xean** [comment]
Merge sometime after Sep4th

**Comment by gomesalexandre** [comment]
Now live: https://polygon.technology/blog/matic-to-pol-migration-is-now-live-everything-you-need-to-know

File: packages/caip/src/adapters/banxa/index.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ const AssetIdToBanxaTickerMap = {
   [optimismAssetId]: 'eth',
   [arbitrumAssetId]: 'eth',
   [baseAssetId]: 'eth',
-  [polygonAssetId]: 'matic',
+  [polygonAssetId]: 'matic', // TODO: check that Banxa update this to POL upstream
```
Comment (gomesalexandre) [inline packages/caip/src/adapters/banxa/index.ts:45]: 💜

File: packages/caip/src/adapters/banxa/index.ts [hunk 2]
```diff
@@ -119,7 +119,7 @@ const chainIdToBanxaBlockchainCodeMap: Record<ChainId, string> = {
   [dogeChainId]: 'DOGE',
   [avalancheChainId]: 'AVAX-C', // note - the AVAX-C chain is not the same as the AVAX "ticker" on the banxa side
   [optimismChainId]: 'OPTIMISM',
-  [polygonChainId]: 'MATIC',
+  [polygonChainId]: 'MATIC', // TODO: check that Banxa update this to POL upstream
```
Comment (gomesalexandre) [inline packages/caip/src/adapters/banxa/index.ts:122]: 💜

File: packages/unchained-client/src/evm/ethereum/parser/__tests__/mockData/tokens.ts [hunk 3]
```diff
@@ -43,10 +43,10 @@ export const tribeToken: Token = {
 }
 
 export const maticToken: Token = {
-  contract: '0x7D1AfA7B718fb893dB30A3aBc0Cfc608AaCfeBB0',
+  contract: '0x455e53CBB86018Ac2B8092FdCd39d8444aFFC3F6',
```
Comment (gomesalexandre) [inline packages/unchained-client/src/evm/ethereum/parser/__tests__/mockData/tokens.ts:46]: ✅

File: scripts/generateAssetData/utils/portals.ts [hunk 4]
```diff
@@ -202,6 +202,8 @@ export const getPortalTokens = async (nativeAsset: Asset): Promise<Asset[]> => {
                 return 'BNB'
               case 'WMATIC':
                 return 'MATIC'
+              case 'WPOL':
+                return 'POL'
```
Comment (gomesalexandre) [inline scripts/generateAssetData/utils/portals.ts:206]: 💜

**Review by gomesalexandre — COMMENTED** [review]
Tested locally:

- POL *token* is happy

image

- POL as a native asset is happy (tested with a full cache clear) 🚫 

image

File: packages/unchained-client/src/evm/ethereum/parser/__tests__/mockData/tokens.ts (hunk 3 above)
Comment (gomesalexandre) [reply packages/unchained-client/src/evm/ethereum/parser/__tests__/mockData/tokens.ts:46]: Actually, we should probably keep the previous flavour here, *or* update to POL, see https://polygontechnology.notion.site/MATIC-POL-Migration-3367ffa954d743b88200cb25cd9f504c

File: packages/unchained-client/src/evm/ethereum/parser/__tests__/ethereum.test.ts [hunk 5]
```diff
@@ -1985,7 +1985,7 @@ describe('parseTx', () => {
       const trade: Trade = { dexName: Dex.Zrx, type: TradeType.Trade }
 
       const buyTransfer = {
-        assetId: 'eip155:1/erc20:0x7d1afa7b718fb893db30a3abc0cfc608aacfebb0',
+        assetId: 'eip155:1/erc20:0x455e53cbb86018ac2b8092fdcd39d8444affc3f6',
```
Comment (gomesalexandre) [inline packages/unchained-client/src/evm/ethereum/parser/__tests__/ethereum.test.ts:1988]: And here too

**Review by gomesalexandre — APPROVED** [review]
Tested after regen in 440cee5b04:

- POL native asset looks sane

image

- POL on mainnet asset looks sane 

image

- Old MATIC token on mainnet is happy

image

- Swaps from legacy MATIC on mainnet are happy

https://jam.dev/c/270c879c-3f4f-4783-9d52-c8f4a0b5d96a

- Swaps from legacy MATIC on mainnet to mainnet POL are happy

https://jam.dev/c/52a8aa2d-79a6-4aed-ba7d-48cba02bb90c

- Swaps to native POL are happy

https://jam.dev/c/eed7f37f-13ea-4b6c-aba5-48bd40149ea7

- Swaps from native POL are happy

https://jam.dev/c/2f22a5c6-e240-4738-a80d-0277b9cc79b2

**Pull request merged**